.madrun.js

```diff
@@ -42,45 +42,79 @@ module.exports = {
     'docker:push:alpine:latest': () => 'docker push coderaiser/cloudcmd:latest-alpine',
-    'docker:build': () => `docker build -t coderaiser/cloudcmd:${version} .`,
-    'docker:build:alpine': () => `docker build -f Dockerfile.alpine -t coderaiser/cloudcmd:${version}-alpine .`,
+    'docker:push:arm32': () => `docker push coderaiser/cloudcmd:${version}-arm32`,
+    'docker:push:arm32:latest': () => 'docker push coderaiser/cloudcmd:latest-arm32',
+    'docker:push:arm64': () => `docker push coderaiser/cloudcmd:${version}-arm64`,
+    'docker:push:arm64:latest': () => 'docker push coderaiser/cloudcmd:latest-arm64',
+    'docker:build': () => `docker build -f docker/Dockerfile -t coderaiser/cloudcmd:${version}-x64 .`,
+    'docker:build:alpine': () => `docker build -f docker/Dockerfile.alpine -t coderaiser/cloudcmd:${version}-alpine .`,
+    'docker:build:arm32': () => `docker build -f docker/arm/Dockerfile.arm32v7 -t coderaiser/cloudcmd:${version}-arm32 .`,
+    'docker:build:arm64': () => `docker build -f docker/arm/Dockerfile.arm64v8 -t coderaiser/cloudcmd:${version}-arm64 .`,
     'docker': () => run(['docker:pull*', 'docker:build*', 'docker:tag*', 'docker:push*']),
     'docker-ci': () => run(['build', 'docker-login', 'docker']),
     'docker-login': () => 'docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD',
-    
+
     'docker:alpine': () => run([
         'docker:pull:alpine',
         'docker:build:alpine',
         'docker:tag:alpine',
         'docker:push:alpine',
         'docker:push:alpine:latest',
     ]),
-    
-    'docker:tag': () => `docker tag coderaiser/cloudcmd:${version} coderaiser/cloudcmd:latest`,
+
+    'docker:arm32': () => run([
+        'docker:pull:arm32',
+        'docker:build:arm32',
+        'docker:tag:arm32',
+        'docker:push:arm32',
+        'docker:push:arm32:latest',
+    ]),
+
+    'docker:arm64': () => run([
+        'docker:pull:arm64',
+        'docker:build:arm64',
+        'docker:tag:arm64',
+        'docker:push:arm64',
+        'docker:push:arm64:latest',
+    ]),
+
+    'docker:manifest': () => run([
+        'docker manifest create coderaiser/cloudcmd:latest coderaiser/cloudcmd:latest-x64 coderaiser/cloudcmd:latest-arm32 coderaiser/cloudcmd:latest-arm64'
+    ]),
+
+    'docker:tag': () => `docker tag coderaiser/cloudcmd:${version}-x64 coderaiser/cloudcmd:latest-x64`,
     'docker:tag:alpine': () => `docker tag coderaiser/cloudcmd:${version}-alpine coderaiser/cloudcmd:latest-alpine`,
-    'docker:rm:version': () => `docker rmi -f coderaiser/cloudcmd:${version}`,
-    'docker:rm:latest': () => 'docker rmi -f coderaiser/cloudcmd:latest',
+    'docker:tag:arm32': () => `docker tag coderaiser/cloudcmd:${version}-arm32 coderaiser/cloudcmd:latest-arm32`,
+    'docker:tag:arm64': () => `docker tag coderaiser/cloudcmd:${version}-arm64 coderaiser/cloudcmd:latest-arm64`,
+    'docker:rm:version': () => `docker rmi -f coderaiser/cloudcmd:${version}-x64`,
```

Comment:
what `x64` suffix used for?

Reply:
The image with the `x64` suffix reflects the current  'regular' cloudcmd image. Since now the image without suffix represents the complete `docker manifest` consisting of the `multi architecture` images

Reply:
According to[ docker documentation](https://docs.docker.com/engine/reference/commandline/manifest/) `manifest` is `experimental` technology and should be enabled editing `config.json`, how can we achive this using [travis](https://travis-ci.org/)? All images build and push using `travis`.

Should every consumer edit it's own `config.json` to pull brand new docker images :)?

Reply:
This needs to be enabled only for the CI in travis. So there is no need to make the adaptions at consumer side. I will research how to make these `config.json` adaptions in travis.

Could you check out the Errors I get regarding the `.madrun.js` [comment](https://github.com/coderaiser/cloudcmd/pull/268#discussion_r365164821)

Reply:
So consumers just use `docker pull coderaisre/cloudcmd` and if they doesn't enable experimental feature, or use old version of `docker` they will receive `coderaiser/cloudcmd:latest` as it was before?

Reply:
Yes, actually when you try to do `docker pull team0/rpi-cloudcmd/` right now. You will get the relevant image pulled. At that exact that time right now, `docker hub` seems down. Otherwise try this [docker-hub-team0-cloudcmd](https://hub.docker.com/r/team0/rpi-cloudcmd/)
The `experimental` `manifest` mode of the docker is only necessary, if you want to `build` and `deploy` the manifest based docker images.

Reply:
I see, `docker pull team0/rpi-cloudcmd`, works good :)

Reply:
>  This needs to be enabled only for the CI in travis. So there is no need to make the adaptions at consumer side. I will research how to make these config.json adaptions in travis.

So what with  `config.json` on `CI` :)? Will it be changed automagically on next release of `Cloud Commander`?

Reply:
Okay just had further research regarding the `Travis CI` topic. It can be adapted throughout some adaption of the `config.json` and `deamon.json`. Therefore I just added a `before_deploy` task in the job which should do the trick.

```
before_deploy:
  - echo '{"experimental":"enabled"}' | sudo tee /etc/docker/daemon.json
  - mkdir -p $HOME/.docker
  - echo '{"experimental":"enabled"}' | sudo tee $HOME/.docker/config.json
  - sudo service docker start
```

[Read here](https://medium.com/@quentin.mcgaw/cross-architecture-docker-builds-with-travis-ci-arm-s390x-etc-8f754e20aaef)

If this all is fine for you, we can maybe resolve everything :)

Reply:
@coderaiser What about resolving this? :)